### PR TITLE
Webhook to validate []genruntime.SecretDestination

### PR DESCRIPTION
**Pull request by matthchr**
**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

**Review by babbageclunk — APPROVED** [review]
Looks great! Minor suggestion about the implementation of the validation function but no biggie.

File: v2/pkg/genruntime/secrets.go [hunk 1]
```diff
@@ -76,3 +80,32 @@ type SecretDestination struct {
...
 }
+
+func (s SecretDestination) String() string {
+	return fmt.Sprintf("Name: %q, Key: %q", s.Name, s.Key)
+}
+
+// ValidateSecretDestinations checks that no destination is writing to the same secret/key, as that could cause
+// those secrets to overwrite one another.
+func ValidateSecretDestinations(destinations []*SecretDestination) error {
+	// Map of secret -> keys
```
[the remark this answers is not in the document]
Comment (matthchr) [reply v2/pkg/genruntime/secrets.go:103]: That's... a lot cleaner.

[the remark this answers is not in the document]
Comment (matthchr) [reply v2/pkg/genruntime/secrets.go:103]: Done.

**Pull request merged**